pkg/operator/ceph/object/status.go

```diff
@@ -76,3 +79,14 @@ func updateStatusBucket(client client.Client, name types.NamespacedName, phase c
 
 	logger.Debugf("object store %q status updated to %v", name, phase)
 }
+
+func buildStatusInfo(cephObjectStore *cephv1.CephObjectStore) map[string]string {
+	m := make(map[string]string)
+	m["endpoint"] = buildDNSEndpoint(BuildDomainName(cephObjectStore.Name, cephObjectStore.Namespace), cephObjectStore.Spec.Gateway.Port, false)
```

Comment:
I still think we need to change the CRD schema to allow the port to be 0 (optional), so it is consistent with the code, so we would need an optional check for `port` here as well.

Reply:
We should probably open an issue for that, right?

Reply:
Even the code makes `Port` mandatory and `securePort` optional. Where do you see it differently? https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/rgw.go#L321

Reply:
The port looks optional in [this check](https://github.com/rook/rook/blob/eabedd7f79dd6602ab1f011693096d54cf39c697/pkg/operator/ceph/object/spec.go#L208), but in other places it is mandatory. So I'll open a new issue to address this separately.

Reply:
See #5806